gax/src/test/java/com/google/api/gax/batching/BatcherImplTest.java

```diff
@@ -711,6 +713,123 @@ public void run() {
     }
   }
 
+  @Test
+  public void testConstructors() throws InterruptedException {
+    try (BatcherImpl batcher1 =
+        new BatcherImpl<>(
+            SQUARER_BATCHING_DESC_V2,
+            callLabeledIntSquarer,
+            labeledIntList,
+            batchingSettings,
+            EXECUTOR)) {
```

Comment:
Minor: Looks like this long block is repeated in each test (with an extra `flowController` parameter sometimes). It  uses the try-with-resources block, which leads into heavy multiline formatting. Pleas consider putting the creation of this in the private method, so the try blocks can look something like:
```java
try (BatcherImpl batcher1 = createBatcher(null)) {
}
// ...

try (BatcherImpl batcher2 = createBatcher(flowController)) {
}
```
That will let save quite a few lines of code here and in general make these tests less scary-looking.